### PR TITLE
Added sign certificate functionality of PKI secret engine. 

### PR DESCRIPTION
**Pull request by slutkiewicz**
PKI secret engine sign certificate endpoint implementation.

### CODE REVIEW

File: src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs [hunk 1]
```diff
@@ -24,6 +24,17 @@ public async Task<Secret<CertificateCredentials>> GetCredentialsAsync(string pki
...
             return result;
         }
+        
+        public async Task<Secret<SignedCertificateData>> SignCertificateAsync(string pkiRoleName, SignCertificatesRequestOptions signCertificatesRequestOptions, string pkiBackendMountPoint = null, string wrapTimeToLive = null)
+        {
+            Checker.NotNull(pkiRoleName, "pkiRoleName");
+            Checker.NotNull(signCertificatesRequestOptions, "signCertificatesRequestOptions");
+
+            var result = await _polymath.MakeVaultApiRequest<Secret<SignedCertificateData>>(pkiBackendMountPoint ?? _polymath.VaultClientSettings.SecretsEngineMountPoints.PKI, "/sign/" + pkiRoleName, HttpMethod.Post, signCertificatesRequestOptions, wrapTimeToLive: wrapTimeToLive).ConfigureAwait(_polymath.VaultClientSettings.ContinueAsyncTasksOnCapturedContext);
+            result.Data.CertificateFormat = signCertificatesRequestOptions.CertificateFormat;
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs:34]: needs a null check on result.Data

File: src/VaultSharp/V1/SecretsEngines/PKI/SignCertificateRequestOptions.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        /// <para>[required]</para>
+        ///  Specifies the PEM-encoded CSR
+        /// </summary>
+        /// <value>
+        ///  Encoded CSR.
+        /// </value>
+        [JsonProperty("csr")]
+        public string Csr { get; set; }
+
+        /// <summary>
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/PKI/SignCertificateRequestOptions.cs:21]: remove this. redundant

File: src/VaultSharp/V1/SecretsEngines/PKI/AbstractCertificateRequestOptions.cs [hunk 3]
```diff
@@ -0,0 +1,113 @@
+using Newtonsoft.Json;
+
+namespace VaultSharp.V1.SecretsEngines.PKI
+{
+    /// <summary>
+    /// Represents the Certificate request options.
+    /// </summary>
+    public abstract class AbstractCertificateRequestOptions
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/PKI/AbstractCertificateRequestOptions.cs:8]: Can you please replicate the class for your need and leave the original POCO intact?

For payload classes, I am deliberately avoiding inheritance to decouple them.

File: src/VaultSharp/V1/SecretsEngines/PKI/CertificateCredentialsRequestOptions.cs [hunk 4]
```diff
@@ -102,24 +21,11 @@ public class CertificateCredentialsRequestOptions
...
-        /// <c>true</c> if [exclude common name from subject alternative names]; otherwise, <c>false</c>.
-        /// </value>
-        [JsonProperty("exclude_cn_from_sans")]
-        public bool ExcludeCommonNameFromSubjectAlternativeNames { get; set; }
-
         /// <summary>
         /// Initializes a new instance of the <see cref="CertificateCredentialsRequestOptions"/> class.
         /// </summary>
-        public CertificateCredentialsRequestOptions()
+        public CertificateCredentialsRequestOptions() : base()
```
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/PKI/CertificateCredentialsRequestOptions.cs:27]: no need for base() right?

File: src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs (hunk 1 above)
Comment (rajanadar) [inline src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs:34]: This property is not marked with JsonIgnore... Either we should let json set it, or if we are explicitly setting it, it should be marked with json-ignore.

**Review by rajanadar — CHANGES_REQUESTED** [review]
minor

File: src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs (hunk 1 above)
Comment (slutkiewicz) [reply src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs:34]: It's assigned as in GetCredentialsAsync function. Thought that there will be some exception, error code from Api if there will be no data send. For my purpose i don't even need that CertificateFormat Assigned cause as a caller you already know it.

Comment (slutkiewicz) [reply src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs:34]: CertificateFormat property is marked as JsonIgnore in AbstractCertificateData class. Or maybe I am referring to wrong property?

File: src/VaultSharp/V1/SecretsEngines/PKI/SignCertificateRequestOptions.cs (hunk 2 above)
Comment (slutkiewicz) [reply src/VaultSharp/V1/SecretsEngines/PKI/SignCertificateRequestOptions.cs:20]: Yes you right. Removed Abstract class and created separate classes for both endpoints so this one is fixed too.

File: src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs (hunk 1 above)
Comment (rajanadar) [reply src/VaultSharp/V1/SecretsEngines/PKI/PKISecretsEngineProvider.cs:34]: got it. i was looking at the request poco by mistake.

**Review by rajanadar — APPROVED** [review]

**Comment by rajanadar** [comment]
thanks @slutkiewicz  for the pr and for using the library

**Comment by rajanadar** [comment]
i'll get this published within 7 days max. maybe faster.

**Comment by rajanadar** [comment]
it was faster.  @slutkiewicz  Your fix is published:

https://www.nuget.org/packages/VaultSharp/1.6.2.3

**Pull request merged**